fl/src/federation/client.py

```diff
@@ -32,29 +34,32 @@ def set_parameters(self, parameters):
 
     def fit(self, parameters, config):
         self.set_parameters(parameters)
-        self.model.current_step = config['current_step']
+        self.model.current_round = config['current_round']
         trainer = Trainer(logger=self.logger, **self.training_params)
         trainer.fit(self.model, datamodule=self.data_module)
-        
         return self.get_parameters(), self.data_module.train_len(), {}
 
+    def calculate_train_loss(self, trainer: Trainer) -> float:
+        loader = self.data_module.predict_dataloader()
+        train_preds = trainer.predict(self.model, loader)
+        y_train = torch.cat([batch[1] for batch in iter(loader)])
+        y_train_pred = torch.cat(train_preds).squeeze(1)
+        # print(f'shapes are: ', y_train.shape, train_preds[0].shape, len(train_preds), y_train_pred.shape)
+        mse = mse_loss(y_train_pred, y_train)
+        return mse.item()
+
     def evaluate(self, parameters, config):
         self.set_parameters(parameters)
 
-        # trainer = Trainer(progress_bar_refresh_rate=0, enable_model_summary=False, logger=self.logger)
         trainer = Trainer(logger=self.logger, **self.training_params)
-        val_loader = self.model.val_dataloader()
-        train_loader = self.model.unshuffled_train_dataloader()
         # train_loader = DataLoader(self.model.train_dataset, batch_size=64, num_workers=1, shuffle=False)
         
-        val_results = trainer.validate(self.model, val_loader, verbose=False)
-        train_results = trainer.validate(self.model, train_loader, verbose=False)
-
-        train_loss, train_accuracy = train_results[0]["val_loss"], train_results[0]["val_accuracy"]
-        val_loss, val_accuracy = val_results[0]["val_loss"], val_results[0]["val_accuracy"]
+        val_results = trainer.validate(self.model, datamodule=self.data_module, verbose=False)
+        val_loss = val_results[0]["val_loss"]
+        train_loss = self.calculate_train_loss(trainer)
 
-        logging.info(f'loss: {train_loss:.4f}\ttrain_acc: {train_accuracy:.4f}\tval_loss: {val_loss:.4f}\tval_acc: {val_accuracy:.4f}')
-        return val_loss, len(val_loader), {"loss": val_loss, "accuracy": val_accuracy}
+        logging.info(f'train_loss: {train_loss:.4f}\tval_loss: {val_loss:.4f}')
+        return val_loss, self.data_module.val_len(), {"val_loss": val_loss, "train_loss": train_loss}
 
 
 def _get_parameters(model):
```

Comment:
a very minor thing but could you move _get_parameters and _set_parameters functions up before the class declaration because they are called in this class?